include/envoy/network/connection_handler.h

```diff
@@ -68,9 +68,67 @@ class ConnectionHandler {
    * after they have been temporarily disabled.
    */
   virtual void enableListeners() PURE;
+
+  /**
+   * Used by ConnectionHandler to manage listeners.
+   */
+  class ActiveListener {
+  public:
+    virtual ~ActiveListener() = default;
+
+    /**
+     * @return the tag value as configured.
+     */
+    virtual uint64_t listenerTag() PURE;
+    /**
+     * @return the actual Listener object.
+     */
+    virtual Listener* listener() PURE;
```

Comment:
Sorry to keep asking for changes here, but why do we need this accessor if we have enable/disable below? Can we have one or the other? Can't we just enable/disable through this?

/wait-any